recipes/sqlite3/all/CMakeLists.txt

```
@@ -165,6 +165,14 @@ if(ENABLE_FTS5 OR ENABLE_MATH_FUNCTIONS)
     endif()
 endif()
 
+# The upstream source files're encoded in UTF-8,
+# but some non-ASCII characters may be not be decoded
+# in the user's current code page (e.g. CP936).
+# Thus we need to instruct the MSVC to use UTF-8.
+if(MSVC)
+    add_compile_options("/utf-8")
+endif()
```

```suggestion
target_compile_options(${PROJECT_NAME} PRIVATE "$<$<C_COMPILER_ID:MSVC>:/utf-8>")
```

Prefer using targets always to not spread the option globally.

---

Updated, thanks

---

Oops, just found that `target_compile_options(${PROJECT_NAME} PRIVATE "$<$<C_COMPILER_ID:MSVC>:/utf-8>")` will not work for the initial problem.

The problematic file `shell.c` belongs to target `sqlite3-bin` not `${PROJECT_NAME}`, so I need to apply the flag to the correct target.

That problem is a bit hard to find. I initially tested your suggestion on the newest version `3.47.0` and everything was fine.
However, the version `3.47.0` does not have the charset issue, so that it's ok without the `/utf-8` flag. The older version e.g. `3.46.1` still suffer, which is the latest one until I make this PR...

So here is a new problem: should we restrict this tweak to specific versions?

---

Good catch, thank you for informing! 

Yes, only the affected version and be applied to sqlite3-bin via target.

---

Fixed